compiler/AST/AstToText.cpp

```diff
@@ -1607,21 +1607,31 @@ void AstToText::appendExpr(CallExpr* expr, bool printingType, const char *outer,
   }
 }
 
+void AstToText::appendExpr(DefExpr* expr, bool printingType, bool isType)
```

Comment:
You never check the value of `printingType` or `isType` in this function.  There isn't a point to sending an argument to a function if it's not going to be used in the function - if you expect it to only be called in certain situations then the function needs to include that in its name so that no one calls your function with isType = false, expecting different behavior

Reply:
I guess I forgot to include that part in the code. I actually thought of first checking for `isType` and then implementing the body, something like
```
void AstToText::appendExpr(DefExpr* expr, bool printingType, bool isType)
{
  if (isType) {
    mText += expr->sym->name;
    if(expr->init) {
      mText += " = ";
      appendExpr(expr->init, printingType);
    }
  }
}
```

Does this look good?

Reply:
I think it would be better to just have it be a helper function with a different name, like `appendExprTypeVar`.  Have that function only be called when `var->isType()` is true, since you're already performing the check.  Don't take in the printingType argument either, just pass true to the appendExpr call like you have on line 1616 on this branch today